library/std/src/sys_common/thread_parker/futex.rs

```diff
@@ -49,7 +49,7 @@ impl Parker {
             // Wait for something to happen, assuming it's still set to PARKED.
             futex_wait(&self.state, PARKED, None);
             // Change NOTIFIED=>EMPTY and return in that case.
-            if self.state.compare_and_swap(NOTIFIED, EMPTY, Acquire) == NOTIFIED {
+            if self.state.compare_exchange_weak(NOTIFIED, EMPTY, Acquire, Acquire).is_ok() {
```

Comment:
I think the Acquire for the failure path is not needed since it'll loop again.

Reply:
As pointed out in the last paragraph of my PR description, I did not want this to turn into optimizing the standard library, just getting rid of `compare_and_swap`. But I think you are right here and I did the change you proposed.

Reply:
I actually would prefer that such changes are split into dedicated PRs - these changes are error prone and I want them easily bisectable if necessary.

Reply:
Yup. That was my initial thought as well. Sorry for getting convinced :P I will back that commit out.

Reply:
Done. I also backed out using `_weak` here completely. So now the updates to `std` should be a 1:1 mapping of what they were before this PR.